lib/srv/sess.go

```diff
@@ -990,6 +990,12 @@ func (s *session) startInteractive(ctx context.Context, ch ssh.Channel, scx *Ser
 			scx.Errorf("Received error waiting for the interactive session %v to finish: %v.", s.id, err)
 		}
 
+		if result != nil {
```

Comment:
I moved this block above the select. The result should be returned before we finish the session. Otherwise, we may miss the exit code.